.github/workflows/ci.yml

```
@@ -87,14 +97,24 @@ jobs:
           version: ${{ env.PROTOC_VERSION }}
           repo-token: '${{ github.token }}'
 
-      - name: Check out Dart Sass
+      - name: Check out Dart Sass only if linked in the PR description
         uses: sass/clone-linked-repo@v1
         with:
           repo: sass/dart-sass
           path: build/dart-sass
           default-ref: null
 
-      - name: Add Dart Sass to pubspec
+      - name: Check out Dart Sass from main HEAD when using -dev dependency
+        run: |
+          if [[ ! -d build/dart-sass ]]; then
+            if grep --quiet -E '\bsass:\s*.*-dev["'"'"']?$' pubspec.yaml; then
```

You can combine these conditions:

```suggestion
          if [[ ! -d build/dart-sass ]] && grep --quiet -E '\bsass:\s*.*-dev["'"'"']?$' pubspec.yaml; then
```